routers/boardRouter.go

```diff
@@ -320,17 +370,633 @@ func (handler *boardHandler) RemoveMemberFromBoard(writer http.ResponseWriter, r
+	params := mux.Vars(request)
+	organizationId := params["organizationId"]
+	boardId := params["boardId"]
+
+	token := request.Context().Value(jwtmiddleware.ContextKey{}).(*validator.ValidatedClaims)
+	tokenCustomClaims := token.CustomClaims.(*auth.CustomClaims)
+	userId := token.RegisteredClaims.Subject
+	readOrgPerm := fmt.Sprintf("org%s:read", organizationId)
+	canReadOrg := tokenCustomClaims.HasPermission(readOrgPerm)
+	if !canReadOrg {
+		http.Error(writer, fmt.Sprintf("User with id %s  does not have permission to read organization with id: %s", userId, organizationId), http.StatusForbidden)
+		return
+	}
+
+	readBoardPerm := fmt.Sprintf("org%s:board%s:read", organizationId, boardId)
+	canReadBoard := tokenCustomClaims.HasPermission(readBoardPerm)
+	if !canReadBoard {
+		http.Error(writer, fmt.Sprintf("User with id %s  does not have permission to read board with id: %s", userId, boardId), http.StatusForbidden)
+		return
+	}
+
+	ctx := request.Context()
+	panels, err := handler.controller.GetPanelsByBoardId(ctx, boardId)
+	if err != nil {
+		http.Error(writer, fmt.Sprintf("Failed to get panels from board with id %s: %s", boardId, err.Error()), http.StatusInternalServerError)
+		return
+	}
+	writer.Header().Set("Content-Type", "application/json")
+	writer.WriteHeader(http.StatusOK)
+	json.NewEncoder(writer).Encode(panels)
+}
+
+func (handler *boardHandler) CreatePanel(writer http.ResponseWriter, request *http.Request) {
+	params := mux.Vars(request)
+	organizationId := params["organizationId"]
+	boardId := params["boardId"]
+
+	title := request.FormValue("title")
+	if title == "" {
+		http.Error(writer, "No Title Found", http.StatusBadRequest)
+		return
+	}
+	token := request.Context().Value(jwtmiddleware.ContextKey{}).(*validator.ValidatedClaims)
+	tokenCustomClaims := token.CustomClaims.(*auth.CustomClaims)
+	userId := token.RegisteredClaims.Subject
+	readOrgPerm := fmt.Sprintf("org%s:read", organizationId)
+	canReadOrg := tokenCustomClaims.HasPermission(readOrgPerm)
+	if !canReadOrg {
+		http.Error(writer, fmt.Sprintf("User with id %s does not have permission to read organization with id: %s", userId, organizationId), http.StatusForbidden)
+		return
+	}
+	createPanelPerm := fmt.Sprintf("org%s:board%s:create_panel", organizationId, boardId)
+	canCreatePanel := tokenCustomClaims.HasPermission(createPanelPerm)
+	if !canCreatePanel {
+		http.Error(writer, fmt.Sprintf("User with id %s does not have permission to create panel on board with id: %s", userId, boardId), http.StatusForbidden)
+		return
+	}
+
+	err := handler.controller.CreatePanel(request.Context(), title, boardId)
+	if err != nil {
+		http.Error(writer, fmt.Sprintf("Failed to create panel: %s", err.Error()), http.StatusInternalServerError)
+		return
+	}
+	writer.Header().Set("Content-Type", "application/json")
+	writer.WriteHeader(http.StatusCreated)
+}
+
+func (handler *boardHandler) GetPanel(writer http.ResponseWriter, request *http.Request) {
```

Comment:
Why did you put these in board router rather than making a panel / card / stack subrouter for their respective methods

Reply:
The reason I chose against that is I felt everything here (panel, card, stack) is strickly related to a board. Like roles are in an org but you can look at an org without looking at its roles. For boards, you are 99% of the time looking at all its content. This also removed nasty importing of models, having them all in the same package made making the `/details` easier. This file is MASSIVE so I understand why we would move them into subrouters but if so I think we need to have subfolders in `routers` to group related routers (users, org & roles, last folder for the rest). lmk what you think

Reply:
but but my functional programming! everything into the smallest categories possible! 

That explanation makes sense. overall i think the difficulty of finding a method in a large file is not a big deal compared to the potential difficulty of moving everything and dealing w the import tax. so i am good with this